src/Fable.AST/Fable.fs

```diff
@@ -409,9 +409,9 @@ type ValueKind =
     | NewArray of newKind: NewArrayKind * typ: Type * kind: ArrayKind
     | NewList of headAndTail: (Expr * Expr) option * typ: Type
     | NewTuple of values: Expr list * isStruct: bool
-    | NewRecord of values: Expr list * ref: EntityRef * genArgs: Type list
+    | NewRecord of values: Expr list * ref: EntityRef * genArgs: Type list * isStruct: bool
```

Comment:
@alexswan10k This is unnecessary, Records and Unions already have entity reference, so you can just check if the entity is `IsValueType` where needed. It needed to be added to anonymous records only because they don't have an entity attached  to them in the F# AST. Please undo this portion of the PR (the change to the AST).